framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -197,6 +196,9 @@
      */
     private final AnnotationFormatter annotationFormatter;
 
+    /** Holds the qualifier upper bounds for type uses. */
+    protected QualifierUpperBounds qualifierUpperBounds;
```

Comment:
Looks like the field could be final.

Reply:
It's initialized in postInit(), so it can't be final.